src/utils/index.ts

```diff
@@ -422,54 +430,26 @@ export const deleteUnique = <T>(array: T[] | undefined, element: T): T[] => {
 export const isEVMWallet = (wallet?: WalletInfo): wallet is EVMWalletInfo => !!wallet && 'connector' in wallet
 export const isSolanaWallet = (wallet?: WalletInfo): wallet is SolanaWalletInfo => !!wallet && 'adapter' in wallet
 
-enum WALLET_KEYS {
-  COIN98 = 'COIN98',
-  BRAVE = 'BRAVE',
-  METAMASK = 'METAMASK',
-  COINBASE = 'COINBASE',
-  TRUST_WALLET = 'TRUST_WALLET',
-  WALLET_CONNECT = 'WALLET_CONNECT',
-}
-
 // https://docs.metamask.io/guide/ethereum-provider.html#basic-usage
 // https://docs.cloud.coinbase.com/wallet-sdk/docs/injected-provider#properties
 // Coin98 and Brave wallet is overriding Metamask. So at a time, there is only 1 exists
-export const detectInjectedType = (): WALLET_KEYS | null => {
-  const { ethereum } = window
-  // When Coinbase wallet connected will inject selectedProvider property and some others props
-  if (ethereum?.selectedProvider) {
-    if (ethereum?.selectedProvider?.isMetaMask) return WALLET_KEYS.METAMASK
-    if (ethereum?.selectedProvider?.isCoinbaseWallet) return WALLET_KEYS.COINBASE
-  }
-
-  if (ethereum?.isCoinbaseWallet) return WALLET_KEYS.COINBASE
-
-  if (ethereum?.isTrustWallet) return WALLET_KEYS.TRUST_WALLET
-
-  if (checkForBraveBrowser() && ethereum?.isBraveWallet) return WALLET_KEYS.BRAVE
-
-  if (ethereum?.isMetaMask) {
-    if (ethereum?.isCoin98) {
-      return WALLET_KEYS.COIN98
-    }
-    return WALLET_KEYS.METAMASK
-  }
-  if (JSON.parse(localStorage.walletconnect || '{}').connected) {
-    return WALLET_KEYS.WALLET_CONNECT
-  }
-  return null
+export const detectInjectedType = (): INJECTED_KEY | undefined => {
+  return INJECTED_KEYS.find(walletKey => {
+    const wallet = SUPPORTED_WALLETS[walletKey]
+    return wallet.readyState() === WalletReadyState.Installed
+  })
 }
 
 export const isOverriddenWallet = (wallet: SUPPORTED_WALLET) => {
   const injectedType = detectInjectedType()
   return (
-    (wallet === WALLET_KEYS.COIN98 && injectedType === WALLET_KEYS.METAMASK) ||
-    (wallet === WALLET_KEYS.METAMASK && injectedType === WALLET_KEYS.COIN98) ||
-    (wallet === WALLET_KEYS.BRAVE && injectedType === WALLET_KEYS.COIN98) ||
-    (wallet === WALLET_KEYS.COIN98 && injectedType === WALLET_KEYS.BRAVE) ||
-    (wallet === WALLET_KEYS.COINBASE && injectedType === WALLET_KEYS.COIN98) ||
+    (wallet === 'COIN98' && injectedType === 'METAMASK') ||
```

Comment:
why don't you add more to enum, but remove all of them 🐳

Reply:
`wallet` is already enum type. `WALLET_KEYS` is a duplicate enum of the existing one. I was remove one and keep one. These `'COIN98'` here is not simply string type but they are enum type already. Dont worry.
![image](https://github.com/KyberNetwork/kyberswap-interface/assets/23650671/9ffab4f8-f33e-45b7-a533-fd88df35bd52)